datajoint/blob.py

```diff
@@ -431,8 +433,16 @@ def read_zero_terminated_string(self):
         self._pos = target + 1
         return data
 
-    def read_value(self, dtype='uint64', count=1):
-        data = np.frombuffer(self._blob, dtype=dtype, count=count, offset=self._pos)
+    def read_value(self, dtype=None, count=1):
+        if dtype is None:
+            dtype = 'uint32' if self.is_32_bit else 'uint64'
+        try:
+            data = np.frombuffer(self._blob, dtype=dtype, count=count, offset=self._pos)
+        except ValueError:
```

Comment:
This may not be a reliable mechanisms for determining the decoding. The serialization should be explicit with respect to how the encoding is performed.

Reply:
@dimitri-yatsenko Yes, we discussed that but the issue is that including the precision into the serialization scheme (e.g. `dj064` or `mym32`) is an update that isn't backward compatible. It would work point forward but wouldn't help us for prior blobs which is precisely where there is likely to be an issue.

Reply:
The Python implementation is deterministic. The serialization does not depend on whether the platform is 32- or 64-bit. 

The problem is that the above solution will not catch all errors. Some cases will not raise the `ValueError`.  We should devise the solution to catch those cases.

Reply:
The real problem is the `mym` implementation depended on the platform. We should explicitly catch this case.

Reply:
We are trying to find an approach with a 'fall-back' mechanism that can capture the vast majority of cases without introducing performance overhead for normal unpacking. This was chosen since one of our customers has an enormous amount of data that would need to be migrated otherwise.

Reply:
As far as I understand, for all numeric arrays, the datatype is fully determined. `mym` underspecified the integer length and it defaulted to the platform default. That was a problem with the matlab implementation only. Correct?  If so, we can have a more targeted fix.